### PR TITLE
Issue #18486: Refractor JavadocParagraph Check

### PR DESCRIPTION
**Pull request by Aman-Baliyan**
Issue: #18486 

what this pr does ?

-> without this pr the check gives false positive for p tags inside block tags now in this pr i added a method to skip the p tags inside the javadoc block tags.

Diff Regression config: https://gist.githubusercontent.com/Aman-Baliyan/c46f95fd657e0ac11ebb04328e11377e/raw/ed493cc08347dcfe016af8e288aade0a8592b86d/JavadocParagraph.xml

### CODE REVIEW

**Comment by Aman-Baliyan** [comment]
is it good that i make a sepaarte method to avoid block tags or otherwise i can merge the logic in method(`isNestedParagraph`) which is used to avoid nested paragraphs ?

**Comment by Aman-Baliyan** [comment]
@romani sir please review

**Comment by romani** [comment]
Please trigger diff testing for this Check.
And confirm explicitly that you reviewed it, and no problem found.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
+
+public class InputJavadocParagraphBlockTag {
+    /**
+     * Test method.
+     *
+     * @return some result
+     *
+     * <p>Indide Block Tag</p>
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: Ok this case is like in issue .

Please add few more test blocks with P tag in block comment

```
* @return some result
*         <p>Indide Block Tag</p>
```
And
```
* @return some result <p>Indide Block Tag</p>
```

**Comment by Aman-Baliyan** [comment]
Github, generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java (hunk 1 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: Please add more test code, as requested here

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: Please add to tests https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/e1076ef_2026182126/reports/diff/guava/index.html#A1

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/e1076ef_2026182126/reports/diff/pmd/index.html#A1
And
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/e1076ef_2026182126/reports/diff/hibernate-orm/index.html#A1

And this (you can shorten text, we don't need much text)
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/e1076ef_2026182126/reports/diff/camunda/index.html#A15

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: Having real examples of usage is very useful, artificial edge case are less valuable.

Comment (Aman-Baliyan) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: > Please add more test code, as requested here

ok sir i will add examples

Comment (Aman-Baliyan) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphBlockTag.java:17]: done
